message.go

```diff
@@ -338,6 +375,74 @@ func (this *CommandMsg) messageUser(UID string, page string, server *Server) {
 	}
 }
 
+func (this *CommandMsg) messageUsers(UIDS []string, page string, server *Server) {
+	msg, err := this.formatMessage()
+	if err != nil {
+		if DEBUG {
+			log.Printf("Error formatting message: %s", err.Error())
+		}
+		return
+	}
+
+	g, ctx := errgroup.WithContext(context.TODO())
+	uc := make(chan map[string]*Socket)
+
+	g.Go(func() error {
+		defer close(uc)
+
+		for _, u := range UIDS {
+			user, err := server.Store.Client(u)
+			if err != nil {
+				if DEBUG {
+					log.Printf("Skipping UID %s because %s", u, err.Error())
+				}
+				return err
+			}
+
+			server.Stats.LogUserMessage()
+
+			select {
+			case uc <- user:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
+
+		return nil
+	})
+
+	g.Go(func() error {
+		for u := range uc {
+			for _, sock := range u {
+				if page != "" && page != sock.Page {
+					if DEBUG {
+						log.Printf("Skipping given page %s != %s", page, sock.Page)
+					}
+
+					continue
+				}
+
+				if !sock.isClosed() {
+					sock.buff <- msg
+				} else {
+					if DEBUG {
+						log.Printf("Skipping because closed")
+					}
+				}
+			}
+
+			select {
+			default:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
+		return nil
+	})
```

Comment:
Rather than spawn two more go routines, what if it looped through the users and called `this.messageUser(uid, page, server)`?